docs/source-2.0/guides/client-guidance/identity-auth/index.md

```diff
@@ -0,0 +1,416 @@
+allows configuration of an IdentityResolver for every auth scheme supported by
+the service.
+
+```java
+public class MyServiceClientConfig {
+    private final IdentityResolver<SigV4CredentialIdentity> sigV4CredentialResolver;
+    private final IdentityResolver<BearerTokenIdentity> bearerTokenResolver;
+
+    // other configuration...
+}
+```
+
+## Signer
+
+A Signer is an entity representing **a way to generate a signature for a
+request**. A signature is metadata attached to a request that will be sent to
+the service in order to allow the service to authenticate the Smithy client
+caller's identity.
+
+For example, an operation that uses a bearer token for
+identification would "sign" an outgoing HTTP request by presenting the value of
+the token in the Authorization header.
+
+All signers require, at a minimum, a request to sign and an identity with which
+to sign. A Signer should be modeled like so:
+
+```java
+// This example interface modifies the transport message in-place with the
+// signature. Depending on the surrounding runtime, the caller may instead wish to
+// model the Signer as an interface that returns a copied, modified transport
+// message instead.
+public interface Signer<TIdentity extends Identity, TMessage extends Message> {
+    void sign(TIdentity identity, TMessage message, Object properties);
+}
+```
+
+## AuthScheme
+
+An AuthScheme self-describes a single flow (combination of an identity resolver
+and signer) through which a Smithy client authenticates a request:
+
+```java
+public interface AuthScheme<TIdentity extends Identity, TMessage extends Message> {
+    /**
+     * Unique identifier for this auth scheme. Typically corresponds to the ID of a
+     * Smithy IDL auth trait, e.g. "smithy.api#httpBearerAuth".
+     */
+	String schemeId();
+
+    /**
+     * Provides an Identity Resolver for this authentication scheme.
+     * This API can return a nullish value indicating that an identity
+     * resolver of this scheme's type is not available in the current client
+     * environment.
+     */
+    IdentityResolver<TIdentity> identityResolver(IdentityResolverConfig config);
+
+    /**
+     * Provides a Signer for this authentication scheme from an unspecified
+     * source. An actual implementation might source the Signer from client
+     * configuration, or perhaps provide its own implementation directly.
+     */
+    Signer<TIdentity, TMessage> signer();
+}
+
+public interface IdentityResolverConfig {
+    IdentityResolver<?> getIdentityResolver(String schemeId);
+}
+```
+
+A list of supported AuthSchemes should be configurable by the end-user of the
+Smithy client. The default "constructor" for a Smithy client should typically
+pre-load a list of supported AuthSchemes on client configuration.
+
+```java
+// in this example, the service supports some combination of
+// smithy.api#httpBearerAuth and aws.auth#sigv4
+public MyServiceClientConfig defaultConfig() {
```

Comment:
These types of code snippets seem useful to the reader but as we build out these docs we want to make sure we have a standard for stuff like this (i.e. showing how to bootstrap a default config).